packages/@css-blocks/glimmer/src/Analyzer.ts

```
@@ -62,15 +62,18 @@ export class GlimmerAnalyzer extends Analyzer<TEMPLATE_TYPE> {
   async analyze(dir: string, componentNames: string[]): Promise<GlimmerAnalyzer> {
     let components = new Set<string>(componentNames);
     this.debug(`Analyzing all templates starting with: ${componentNames}`);
-
-    for (let component of components) {
-      components.add(component);
-      let deps = this.resolver.recursiveDependenciesForTemplate(dir, component);
-      deps.forEach(c => components.add(c));
+    try {
+      for (let component of components) {
+        components.add(component);
+        let deps = this.resolver.recursiveDependenciesForTemplate(dir, component);
+        deps.forEach(c => components.add(c));
+      }
+    } catch (err) {
+      // if for some reason the resolver can't resolve. Otherwise it gets swallowed.
+      throw err;
```

I'm skeptical that this try/catch block is necessary. Any exception raised within the async function should reject the promise... catching and rethrowing it doesn't change anything. Can you add a test case for the issue you're trying to solve with this code?

---

Will test. It certainly helped why I was testing but I wonder if there was a knock-on effect happening with something else. Will see.

---

Can confirm - without it it swallows the error. I only noticed this when I accidentally forced it to load a file that didn't exist. Seems like an edge case but... it seems to silently fail otherwise.

---

I took this change out. It shouldn't be necessary. The tests still pass. I wish I could reproduce the issue you saw...